packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -197,6 +150,21 @@ export const useGiftCards = props => {
         setIsCartUpdating
     ]);
 
+    const shouldDisplayCardBalance = useMemo(
```

Comment:
I would expect this memo to be calculated every time, not sure this is an improvement; what was wrong with the current implementation of derived state?

Reply:
We went on an adventure fixing this bug, it seems the my refactoring to the rest of the talon resolved the original error, so you're correct. The `useMemo` is no longer needed, I've updated the code :)